### PR TITLE
add number format to pivot table

### PR DESCRIPTION
**Pull request by yileic**

### CODE REVIEW

File: superset/assets/javascripts/explorev2/stores/controls.jsx [hunk 1]
```diff
@@ -550,7 +550,8 @@ export const controls = {
     type: 'SelectControl',
     freeForm: true,
     label: 'Number format',
-    default: D3_TIME_FORMAT_OPTIONS[0],
+    renderTrigger: true,
```
Comment (yileic) [inline superset/assets/javascripts/explorev2/stores/controls.jsx]: if use 'D3_TIME_FORMAT_OPTIONS[0]', Number format field is blank in the UI somehow.

File: superset/assets/javascripts/explorev2/stores/controls.jsx [hunk 2]
```diff
@@ -550,7 +550,8 @@ export const controls = {
...
+    default: '.3s',
```
Comment (yileic) [inline superset/assets/javascripts/explorev2/stores/controls.jsx]: if use 'D3_TIME_FORMAT_OPTIONS[0]', Number format field is blank in the UI somehow.

**Comment by mistercrunch** [comment]
For reference, the way that this is done for `table.js` is by leveraging the metric's `d3format` configuration through the `slice.d3format` method.

You can imagine that people using pivot table may want different format for different metrics. 

The longer term solution for this is in a branch on my repo `mistercrunch/metrics` where there's a new intricate control introduced for metrics, where people would define metrics in the explore view as either an expression, agg function/column, or by referencing a predefined metric in a popover, all of that in a popover that also allows labeling of the metric and formatting.

**Comment by mistercrunch** [comment]
We may want to merge that in the meantime, it would work for pivoting a single metric or many metrics where the same formatting applies.

**Comment by yileic** [comment]
I see. Thank you so much for the insight!

**Comment by mistercrunch** [comment]
Merging for now. There may be some non-100% backward compatible changes in the future, but it should be very minor.

**Pull request merged**